app/models/power_relay.rb

```diff
@@ -1,15 +1,18 @@
 # frozen_string_literal: true
 
 # Including class must implement #relay_connection
-# Port refers to the outlet, not the IP port
 module PowerRelay
 
   extend ActiveSupport::Concern
 
+  MAXIMUM_NUMBER_OF_OUTLETS = 1..16.freeze
+
   included do
     ## validations
-    validates_presence_of :ip, :port, :username, :password
+    validates_presence_of :ip, :outlet, :username, :password
     validates :auto_logout_minutes, presence: { if: :auto_logout }
+    validates :outlet, inclusion: MAXIMUM_NUMBER_OF_OUTLETS
```

Comment:
What would you think of using a `numericality` validation? Since the db field is an integer, that feels like it might be more consistent.

And this is kind of a personal preference, but I lean towards putting all the validations on a field together.

```
validates :outlet, presence: true, numericality: { only_integer: true, greater_than: 0, less_than_or_equal_to: 16 }
```

Reply:
sure, works for me